lib/proxy/log_buffer/buffer.rb

```diff
@@ -0,0 +1,83 @@
+require 'proxy/log_buffer/ring_buffer'
+
+module Proxy::LogBuffer
+
+  LogRecord = Struct.new(:serial, :timestamp, :level, :message)
+  BacktraceLogRecord = Struct.new(:serial, :timestamp, :level, :message, :backtrace)
+
+  class Buffer
+    def self.instance
+      @@buffer ||= Buffer.new
+    end
+
+    def initialize
+      @mutex = Mutex.new
+      @serial = 0
+      @buffers_set_up = false
+    end
+
+    def new_serial
+      @serial += 1
+    end
+
+    # late initialization
+    def setup_buffers(size = nil, size_tail = nil, level = nil, level_tail = nil)
```

Comment:
why delay the initialization? Our startup times aren't slow, this does delay potential issues, making them harder to troubleshoot.